Sources/KituraStencil/KituraStencilTemplateEngine.swift

```diff
@@ -41,23 +41,82 @@ public enum StencilTemplateEngineError: Swift.Error {
     case noKeyProvidedForType(value: Encodable)
 }
 
+/**
+ A `TemplateEngine` for Kitura that uses Stencil for templating.
+
+ The default file extension for templates using this engine is `stencil`. If you do
+ not explicitly provide a file extension in the call to `response.render` then this
+ extension will be applied automatically.
+
+ ### Usage Example: ###
+ ```swift
+    router.add(templateEngine: StencilTemplateEngine())
+
+    // An example of using a dictionary of [String: Any] parameters to be rendered
+    router.get("/hello") { request, response, next in
+        try response.render("StencilExample.stencil", context: ["name": "World!"]])
+        next()
+    }
+
+    // A codable type containing structured data to be used in our template
+    struct Friend: Codable {
+        let firstName: String
+        let lastName: String
+    }
+
+    // Structured data that we wish to render
+    let friends = [Friend(firstName: "Jack", lastName: "Sparrow"), Friend(firstName: "Captain", lastName: "America")]
+
+    // An example of using type-safe templating to render data from a Swift type
+    router.get("/friends") { request, response, next in
+        try response.render("MyStencil.stencil", with: friends, forKey: "friends")
+        next()
+    }
+ ```
+ For more information on type-safe templating, see: https://developer.ibm.com/swift/2018/05/31/type-safe-templating/
+ */
 public class StencilTemplateEngine: TemplateEngine {
+
+    /// The file extension of files rendered by the KituraStencil template engine: `stencil`
     public let fileExtension = "stencil"
+
     private let `extension`: Extension
     private var rootPaths: [Path] = []
 
+    /// Initializes a KituraStencil template engine.
+    ///
+    /// - Parameter extension: An optional Stencil `Extension` for customizing the
+    ///   underlying template engine.
     public init(extension: Extension = Extension()) {
         self.`extension` = `extension`
     }
 
+    /// Set root paths for the template engine - the paths where the included templates can be
```

Comment:
"the paths where the included templates can be searched" - I don't like this wording - same as in the Parameter definition below.